mark missing steps in navbar

josihoppe opened the pull request:
Fixes #77 

I added the classes "incomplete" and "complete" as options for the sidebar. 
complete is for flows, that are finished, incomplete for flows that are not finished
"active" is still working the same, and visited is used right now for views and categories. 

the incomplete/complete classes are overwriting the visited state right now. So once a flow is completet/finished it will be shown as that as long as it stays finished. That results in complete flows where they might just be shown as not_visited before, was not sure if thats the desired behaviour.

Also replaced the plain Flow names with the labels/names from the sidebar for easier readability. 

I just used different colors for incomplete and complete for testing at the moment.

josihoppe commented:
Just saw, that having only the complete/incomplete categories for the flows results in financial support to always be shown as complete since its not required to enter information there. Should this behaviour be changed?

nesnoj commented:
Thx @josihoppe! However I cannot test: I deleted my session data and started from the beginning but getting "KeyError at /intro_inventory/" -> 'building_type' once I try to proceed..

josihoppe commented:
> Thx @josihoppe! However I cannot test: I deleted my session data and started from the beginning but getting "KeyError at /intro_inventory/" -> 'building_type' once I try to proceed..

Found the cause of the error and I think I fixed it. :)

nesnoj commented:
Great, looks good!
And yes, the current behavior in financial support is ok.

One change request: To my mind it makes more sense to have the start optimization as extra main entry like this:

![image](https://github.com/user-attachments/assets/c012e3e9-ecc9-4ba3-874e-432591b44248)

Could you please change this?

bmlancien commented:
@nesnoj Is this working for you?

nesnoj commented:
> @nesnoj Is this working for you?

Yes, thx!

I find the red color a bit too striking (the user has not even started to give any input and sees red) and would prefer to have it a little more unobtrusive. What about dropping the color and just use filled/unfilled version like in my last commit? What do you think @bmlancien?

![image](https://github.com/user-attachments/assets/b950ceaf-9b30-4eee-b990-22858e49790d)

bmlancien commented:
I think that's because I'm not totally sure how the flow is working. I used red for when a user has been through a step without adding all mandatory input and is now in a further step (I even thought this case was not possible). In the case the user hasn't started to give any input yet, then of course it shouldn't be red, there shouldn't be any warning actually. Keeping it black like now is better.

nesnoj commented:
> I think that's because I'm not totally sure how the flow is working. I used red for when a user has been through a step without adding all mandatory input and is now in a further step (I even thought this case was not possible). In the case the user hasn't started to give any input yet, then of course it shouldn't be red, there shouldn't be any warning actually. Keeping it black like now is better.

Alright, then we keep it like this, thank you!
